swarm/network/stream/stream.go

```diff
@@ -138,7 +143,7 @@ func (r *Registry) GetServerFunc(stream string) (func(*Peer, []byte, bool) (Serv
 
 func (r *Registry) RequestSubscription(peerId discover.NodeID, s Stream, h *Range, prio uint8) error {
```

Comment:
I am not sure I understand the changes to this method. This doesn't appear to have anything to do with starting to sync when the depth changed on Kademlia. @janos could you maybe add a short description to the PR and explain the changes in this method? Thanks :)

Reply:
The changes in this method are basically described in a comment "request subscription only if the server for this stream is not created". If there is already a server for the stream, not to try to request a subscription again. The relation with kademlia depth change is when the startSyncing is called on depth change, it will ensure that no request subscriptions are made if the server already exists.

Reply:
OK, thanks :)